### PR TITLE
Migrate to rattler-build

### PR DESCRIPTION
**Pull request by cbourjau**
This PR migrates this feedstock to use rattler-build.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+tests:
+  - python:
+      imports:
+        - spox
+  - script:
+      - pip check
+    requirements:
+      run:
+        - pip
```
Comment (xhochy) [inline recipe/recipe.yaml]: You don't need it, it is default behaviour included in the `python` test.
```suggestion
```

**Comment by cbourjau** [comment]
As discussed internally, there shouldn't be three individual NumPy jobs. Right now, it is unclear to me where those are coming from.

**Comment by cbourjau** [comment]
@conda-forge-admin please rerender

**Comment by cbourjau** [comment]
The issue with the three NumPy jobs appears to have been solved upstream. CI looks fine to me now.

**Review by xhochy — APPROVED** [review]

**Pull request merged**